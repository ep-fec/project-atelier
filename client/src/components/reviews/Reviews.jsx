import React, { useState, useEffect } from 'react';
import List from './components/List.jsx';
import Review from './components/Review.jsx';
import ProductBreakdown from './components/ProductBreakdown.jsx';
import Ratings from './components/Ratings.jsx';
import Sort from './components/Sort.jsx';
import NewReview from './components/NewReview.jsx';
import data from './sampleData.js';
import axios from 'axios';

const Reviews = (props) => {

  const getReviews = () => {
    axios.get(`/reviews?product_id=${props.currentProduct}&count=50&sort=newest`)
      .then((res) => setReviews(res.data))
      .catch((err) => console.log(err));
  }

  let [reviews, setReviews] = useState({results: []});

  useEffect(() => {
    getReviews();
  }, [props.currentProduct]);

  return (
    <section className="reviews-container">
      <h2>RATINGS & REVIEWS </h2>
      <br/><br/>

      <section className="reviews leftcol">
        <Ratings />
        <ProductBreakdown />
      </section>

      <section className="reviews rightcol">
        <Sort reviews={reviews}/>
        <List reviews={reviews}/>
        <br/>
        <button className="reviews more-reviews reviewsbutton">MORE REVIEWS</button>
        <button className="reviews add-review reviewsbutton">ADD A REVIEW +</button>
      </section>
    </section>
  )
}

export default Reviews;