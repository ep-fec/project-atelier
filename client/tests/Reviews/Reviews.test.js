import renderer from 'react-test-renderer';
import Reviews from '../../src/components/reviews/Reviews.jsx';

it('Renders the initial Reviews component correctly', async () => {
  const currentProduct = {id: 71697}
  const tree = await renderer.create(<Reviews currentProduct={currentProduct}/>).toJSON();
  await expect(tree).toMatchSnapshot();
});