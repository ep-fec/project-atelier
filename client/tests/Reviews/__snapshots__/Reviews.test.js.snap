// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the initial Reviews component correctly 1`] = `
<section
  className="reviews-container"
>
  <h2>
    RATINGS & REVIEWS 
  </h2>
  <br />
  <br />
  <section
    className="reviews leftcol"
  >
    <div
      className="reviews rating-breakdown"
    >
      Rating Breakdown
    </div>
    <div
      className="reviews product-breakdown"
    >
      Product Breakdown
    </div>
  </section>
  <section
    className="reviews rightcol"
  >
    <div
      className="reviews sort-options"
    >
      -Sort Options Placeholder-
    </div>
    <div
      className="reviews reviews-list"
    >
      <div
        className="reviews no-reviews"
      >
        <button
          className="reviews add-review-centered reviewsbutton"
        >
          ADD A REVIEW +
        </button>
      </div>
      <div
        className="reviews reviews-list-extended"
        style={
          Object {
            "--max-height": "350px",
          }
        }
      >
        <br />
      </div>
    </div>
  </section>
</section>
`;
